extensions/amp-iframe/1.0/component.js

```diff
@@ -26,23 +30,96 @@ export function Iframe({
   allowFullScreen,
   allowPaymentRequest,
   allowTransparency,
-  onLoadCallback = NOOP,
+  onLoad = NOOP,
   referrerPolicy,
+  requestResize,
   sandbox,
   src,
   srcdoc,
   ...rest
 }) {
+  const iframeRef = useRef();
+  const dataRef = useRef(null);
+  const isIntersectingRef = useRef(false);
+
+  const attemptResize = useCallback(() => {
+    const iframe = iframeRef.current;
+    if (!iframe) {
+      return;
+    }
+    const height = Number(dataRef.current.height);
+    const width = Number(dataRef.current.width);
+    if (!height && !width) {
+      return;
+    }
+    if (requestResize) {
+      // Currently `requestResize` is called twice:
+      // 1. when the message is received in viewport
+      // 2. when exiting viewport
+      // This could be optimized by reducing to one call by assessing when to call.
+      requestResize(height, width);
+      iframe.height = FULL_HEIGHT;
```

Comment:
Why are we changing the iframe size here?

Reply:
See response to https://github.com/ampproject/amphtml/pull/35566/files#r692360292 above.

Reply:
requestResize can be async. Until the promise succeeds, changing this value can actually cause the iframe to collapse.

Reply:
Good point. Updated to return promise.